### DISCUSSION TITLE
Makerfabs ESP32 3.5" TFT Touch

### DISCUSSION THREAD

**Discussion by mageslain** [Bodmer/TFT_eSPI#1229] [Compatible displays and setup files]
Hi,
Has anyone had any luck with this esp/touchscreen?
https://www.makerfabs.com/esp32-3.5-inch-tft-touch-capacitive-with-camera.html

I've seen a few issues with the ILI9488 screen. From my initial test all I got was a white screen.

**Comment by mageslain** [comment]
I got it working just had to set my pins as so: 
#define TFT_MISO 12
#define TFT_MOSI 13
#define TFT_SCLK 14
#define TFT_CS   15
#define TFT_DC   33
#define TFT_RST   3

I haven't looked into the touch screen aspect of it yet.

**Comment by jack828** [comment]
I have the exact same hardware! However, I can't get any output on my screen. Do you mind posting your `Read_User_Setup.ino` or some more config please?

Here's mine:
```
TFT_eSPI ver = 2.3.70
Processor    = ESP32
Frequency    = 240MHz
Transactions = Yes
Interface    = SPI
Display driver = 9341
Display width  = 240
Display height = 320

MOSI    = GPIO 13
MISO    = GPIO 12
SCK     = GPIO 14
TFT_CS   = GPIO 15
TFT_DC   = GPIO 33
TFT_RST  = GPIO 3

Font GLCD   loaded
Font 2      loaded
Font 4      loaded
Font 6      loaded
Font 7      loaded
Font 8      loaded
Smooth font enabled

Display SPI frequency = 40.00
```

EDIT: I see that `Display driver = 9341` when I have enabled `ILI9488_DRIVER` ?
EDIT2: I found the schematic.

[ESP32_TFT_Touch_with_camera(3.5''_ili9488)_sch.pdf](https://github.com/Bodmer/TFT_eSPI/files/6848289/ESP32_TFT_Touch_with_camera.3.5._ili9488._sch.pdf)


RE: the touch screen - looks like it isn't supported so you can't use this library for that aspect. https://github.com/Bodmer/TFT_eSPI/issues/152#issuecomment-395205036

**Reply by jack828** [reply]
Finally got it.

Turns out that `Display driver = 9341` output was the kicker. I thought that I'd need only uncomment a pre-set up file and I'd be off, but I first had to comment out the `User_setup.h` as it was defining the wrong driver!

For reference, my setup file was:
```h
#define ILI9488_DRIVER

#define TFT_MISO 12
#define TFT_MOSI 13
#define TFT_SCLK 14
#define TFT_CS 15
#define TFT_DC 33
#define TFT_RST 26


#define LOAD_GLCD   // Font 1. Original Adafruit 8 pixel font needs ~1820 bytes in FLASH
#define LOAD_FONT2  // Font 2. Small 16 pixel high font, needs ~3534 bytes in FLASH, 96 characters
#define LOAD_FONT4  // Font 4. Medium 26 pixel high font, needs ~5848 bytes in FLASH, 96 characters
#define LOAD_FONT6  // Font 6. Large 48 pixel font, needs ~2666 bytes in FLASH, only characters 1234567890:-.apm
#define LOAD_FONT7  // Font 7. 7 segment 48 pixel font, needs ~2438 bytes in FLASH, only characters 1234567890:.
#define LOAD_FONT8  // Font 8. Large 75 pixel font needs ~3256 bytes in FLASH, only characters 1234567890:-.
#define LOAD_GFXFF  // FreeFonts. Include access to the 48 Adafruit_GFX free fonts FF1 to FF48 and custom fonts

#define SMOOTH_FONT


#define SPI_FREQUENCY       40000000
#define SPI_READ_FREQUENCY  16000000
#define SPI_TOUCH_FREQUENCY  2500000

#define USE_HSPI_PORT
```

Will keep this thread updated if I figure out the touch display too.

**Reply by mageslain** [reply]
Ah cool, glad you figured it out. I reformatted my computer without backing up my arduino librarys so I've lost my settings anyway haha.
I've since switched to use the lovyanGFX library since the documentation for our screens seems to use that library more often. Touch was already setup for that library so was easy to implement.

**Reply by jack828** [reply]
No sweat. I almost gave up and used that library but wanted to use this one (mainly because it's in English 😄). I've kept the touchscreen code so hoping I can get that to work relatively easy with what this lib currently supports - might even submit a patch if I'm feeling brave.

Best of luck in your project!
